### PR TITLE
fix: add a11y linting tools and fix errors

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses #1825

- [x] This change addresses only certain aspects of the issue

## Description

This PR adds 1 of 2 of the a11y linting tools in the issue. I separated them out since there were ~40 errors the first linting tool flagged.

## How Can This Be Tested/Reviewed?

Most of the changes are moving actionable pieces to be buttons, which has the potential to cause some visual changes because of different styling applied, so I checked out the cases where there were tag changes in the app. In some cases there were changes I kept because it actually made us visually more consistent, since the errors we had were outliers for us.

@slowbot Would love to review this with you to confirm the few slight visual changes.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/src/LinkComponent.tsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ const LinkComponent = (props: PropsWithChildren<LinkProps>) => {
   const anchorProps = { ...props }
   delete anchorProps.href
 
+  /* eslint-disable jsx-a11y/anchor-has-content */
```
Comment (emilyjablonski) [inline sites/public/src/LinkComponent.tsx:9]: This is the only rule that I disabled in both `LinkComponent` files in public and partners, which I think makes sense for our use here

File: ui-components/src/sections/GridSection.tsx [hunk 2]
```diff
@@ -22,7 +22,6 @@ const GridCell = (props: GridCellProps) => {
 
 export interface GridSectionProps {
   title?: React.ReactNode
-  edit?: string
```
Comment (emilyjablonski) [inline ui-components/src/sections/GridSection.tsx:25]: We don't use this anywhere

**Comment by emilyjablonski** [comment]
The AccessLint job is still pending, but I think it's because it was failing and is now passing and it got stuck ¯\_(ツ)_/¯

**Review by ludtkemorgan — APPROVED** [review]
Looks good, thanks for doing this. I probably missed some of the places while attempting to QA, but everywhere I did it looked good.

File: sites/partners/pages/settings/index.tsx [hunk 3]
```diff
@@ -38,25 +38,25 @@ const Settings = () => {
     return (
       <div className={"flex justify-end"}>
         <div className={"w-max"}>
-          <span onClick={() => setPreferenceDrawerOpen(true)} className={"cursor-pointer"}>
+          <button onClick={() => setPreferenceDrawerOpen(true)} className={"cursor-pointer"}>
```
[the remark this answers is not in the document]
Comment (emilyjablonski) [reply sites/partners/pages/settings/index.tsx]: It isn't good call! Removed 👍

**Comment by emilyjablonski** [comment]
Screen Shot 2022-08-19 at 4 24 23 PM
Screen Shot 2022-08-19 at 4 24 35 PM

File: ui-components/src/sections/GridSection.tsx (hunk 2 above)
Comment (seanmalbert) [reply ui-components/src/sections/GridSection.tsx:25]: Does DAHLIA?

**Review by seanmalbert — APPROVED** [review]
lgtm, @emilyjablonski. There's the question if DAHLIA uses the edit param for GridSection. 

This is more of a note, another thing that stood out to me while reviewing was the use of braces for strings like `aria-label={"Launch video"}` vs `aria-label="Launch video"`. My initial thought was that the braces may cause unnecessary processing, but looking into it, it seems like it's probably trivial.

**Comment by emilyjablonski** [comment]
@seanmalbert Good q, SF doesn't use this component afaik 👍

**Pull request merged**